### PR TITLE
feat(grep): Jump to next file match segment

### LINKED ISSUE #512: [FEATURE REQUEST] fff.nvim - Choose specific file in grep
Hi!
I am a frequent user of the grep functionality and sometimes there's a lot of results before the file I want to get to. 

It would be really nice if you could specify a specific file like `main.py`, and not only `*.py`.

I am a bit unsure how this could be implemented not to conflict with preexisting search modes, especially in the fuzzy, but perhaps in the plain search mode or a prefix to choose a specific file. 

A simple example shown below in a small repo. There are some ways to get around mashing arrow down or <C-n> but a built in way to grep a specific file or quickly choose the instance from the list would be really nice. 

Image

Image

### PR DESCRIPTION
**Pull request by Hex4C**
Fully closes #512 

---

Heavily inspired by #532 (squashed based on that PR)

Since a majority of the logic already was implemented and only needed to be slightly adapted to the new structure for the picker_ui I wrote a majority of the code with LLM assisted tools.

I did the same thing for the unit test, but I am less certain of them... They seemed slightly dated, they can almost certainly be improved.

I have tested the UI for edge cases such as no files, 1 file and multiple files etc and it has worked flawlessly so far.

Feedback is greatly appreciated or edit the files directly if you wish. Hope this helps!

### CODE REVIEW

**Comment by Hex4C** [comment]
Force pushed to remove picker_ui.lua (original before refactor), accidentally added it.

**Comment by Hex4C** [comment]
Messed up the test setup, fixed now and passes the 4 unit tests.

File: lua/fff/picker_ui/navigation.lua [hunk 1]
```diff
@@ -117,7 +117,7 @@ function M.move_up()
 
   if not P.render_after_cursor_move(old_cursor) then return end
   P.update_status()
-  pcall(vim.cmd, 'redraw')
```
Comment (dmtrKovalenko) [inline lua/fff/picker_ui/navigation.lua:120]: why did you remove pcalls? They are there on purpose

Comment (Hex4C) [reply lua/fff/picker_ui/navigation.lua:120]: Whoops my bad, had some lsp warnings and wanted to use vim.cmd.redraw to avoid the type mismatch warning. Meant to use *pcall*(vim.cmd.redraw), not sure though if it is exactly the same so I'll leave as it was before (with pcall(vim.cmd, 'redraw'))
